spec/unit/defaults_spec.rb

```diff
@@ -234,4 +234,21 @@
       Puppet.initialize_settings
     end
   end
+
+  describe "the default cadir", :unless => Puppet::Util::Platform.windows?  do
+    it 'defaults to inside the ssldir if not migrated' do
+      expect(File).to receive(:exist?).with('/etc/puppetlabs/puppetserver/ca/ca_crt.pem').and_return(false)
+      expect(Puppet.default_cadir).to eq("#{Puppet[:ssldir]}/ca")
+    end
+
+    it 'returns the new location if the old location is a symlink' do
+      expect(File).to receive(:exist?).with('/etc/puppetlabs/puppetserver/ca/ca_crt.pem').and_return(true)
+      expect(Puppet.default_cadir).to eq('/etc/puppetlabs/puppetserver/ca')
+    end
+
+    it 'returns an empty string for Windows platforms', :if => Puppet::Util::Platform.windows? do
+      Puppet.initialize_settings
```

Comment:
Is the `initialize_settings` actually needed here? It's not called in the other two tests...

Reply:
I stopped calling it because there was weird caching going on, and I felt like testing the method directly was better. But I forgot to update this one because it doesn't run for me :P Thanks for catching this.

Reply:
updated